js/battle.js

```diff
@@ -5566,6 +5563,7 @@ var Battle = (function () {
 		}
 	};
 	Battle.prototype.run = function (str, preempt) {
+		var takesKwargs = {swap:1, move:1, cant:1, callback:1};
```

Comment:
Sounds reasonable, this would help implementing extra kwargs while preserving the compatibility. I would fix that... but I lately don't have much time, although that should be dealt with in this week.